src/Illuminate/Collections/Json.php

```diff
@@ -0,0 +1,255 @@
+<?php
+
+namespace Illuminate\Support;
+
+use ArrayAccess;
+use Illuminate\Contracts\Support\Arrayable;
+use Illuminate\Contracts\Support\Jsonable;
+use Illuminate\Support\Traits\Conditionable;
+use Illuminate\Support\Traits\Tappable;
+use JsonSerializable;
+use Stringable;
+
+class Json implements ArrayAccess, Arrayable, JsonSerializable, Jsonable, Stringable
+{
+    use Conditionable;
+    use Tappable;
+
+    /**
+     * The underlying JSON data.
+     *
+     * @var array
+     */
+    protected $data;
+
+    /**
+     * Create a new Json instance.
+     *
+     * @param  iterable  $data
+     */
+    public function __construct($data = [])
+    {
+        foreach ($data as $key => $value) {
+            $this->data[$key] = $value;
+        }
```

Comment:
Do you need to map through the entire array here? 
The only thing I can think is that if the array has a duplicated key, then you will keep the last one, but it already happens even if you just assign the value using `$this->data = $data`.

Check this example. Using foreach:
```php
class Json {
    public array $data;
  
    public function __construct($data = []) {
        foreach($data as $key => $value) {
            $this->data[$key] = $value;
        }
  }
}

$data = ['test' => 'a', 'b' => 'c', 'test' => 1, 'test' => 2]; 

new Json($data)

/*
This code outputs
Json {#2885
     +data: [
       "test" => 2,
       "b" => "c",
     ],
   }
*/
``` 

Using `$this->data = $data`:
```php
class Json {
    public array $data;
  
    public function __construct($data = []) {
        $this->data = $data;
   }
}

$data = ['test' => 'a', 'b' => 'c', 'test' => 1, 'test' => 2]; 

new Json($data)

/*
This code outputs

Json {#2885
     +data: [
       "test" => 2,
       "b" => "c",
     ],
   }
*/
```

Reply:
It's the same approach the `Fluent` class. 

Since you can issue an `iterable` object (like something `IteratorAggregate`), this ensures all the items are rooted into an `array`. Otherwise, I could copy how the `Collection` does it, which does not map the entire array but rather checks which is:

```php
if (is_array($items)) {
    return $items;
} elseif ($items instanceof Enumerable) {
    return $items->all();
} elseif ($items instanceof Arrayable) {
    return $items->toArray();
} elseif ($items instanceof Jsonable) {
    return json_decode($items->toJson(), true);
} elseif ($items instanceof JsonSerializable) {
    return (array) $items->jsonSerialize();
} elseif ($items instanceof Traversable) {
    return iterator_to_array($items);
}

return (array) $items;
```

Reply:
Got it. Thanks!

Reply:
I said "I could", not that I will add it to the PR because, sincerely, I think it adds more boilerplate.